lib/NMISNG/Auth.pm

```diff
@@ -2141,4 +2199,383 @@ sub update_last_login
+	my $auth_ldap_psw        = $self->{config}->{'auth_ldap_psw'};
+	my $auth_ldap_server     = $self->{config}->{'auth_ldap_server'};
+	my $auth_ldaps_server    = $self->{config}->{'auth_ldaps_server'};
+	if ( $self->{auth} eq "ms-ldap" or $self->{auth} eq "ms-ldaps") {
+		NMISNG::Util::logAuth("INFO Honoring legacy ActiveDirectory settings.") if ($self->{debug});
+		# Prioritize 'auth_ldap_psw' over 'auth_ms_ldap_dn_psw' if both are defined
+		# Use 'auth_ldap_psw' if available, otherwise fallback to 'auth_ms_ldap_dn_psw'. Remember 'auth_ms_ldap_dn_psw' is retired value.
+		if (defined ($self->{config}->{'auth_ms_ldap_psw'}) || defined ($self->{config}->{'auth_ms_ldap_dn_psw'})){
+			$auth_ldap_psw = NMISNG::Util::decrypt($self->{config}->{auth_ms_ldap_psw}, 'authentication', 'auth_ms_ldap_psw') // $self->{config}->{'auth_ms_ldap_dn_psw'};
+		}
+		# Retrieve the LDAP base configuration
+		# Prioritize 'auth_ms_ldap_base' over 'auth_ms_ldap_context' if both are defined
+		# Use 'auth_ms_ldap_base' if available, otherwise fallback to 'auth_ms_ldap_context'. Remember 'auth_ldap_context' is retired value.
+		if (defined ($self->{config}->{'auth_ms_ldap_base'}) || defined ($self->{config}->{'auth_ms_ldap_context'})){
+			$auth_ldap_base = $self->{config}->{'auth_ms_ldap_base'} // $self->{config}->{'auth_ms_ldap_context'};
+		}
+		# Prioritize 'auth_ms_ldap_acc' over 'auth_ms_ldap_dn_acc' if both are defined
+		# Use 'auth_ms_ldap_acc' if available, otherwise fallback to 'auth_ms_ldap_dn_acc'. Remember 'auth_ms_ldap_dn_acc' is retired value.
+		if (defined ($self->{config}->{'auth_ms_ldap_acc'}) || defined ($self->{config}->{'auth_ms_ldap_dn_acc'})){
+			$auth_ldap_acc = $self->{config}->{'auth_ms_ldap_acc'} // $self->{config}->{'auth_ms_ldap_dn_acc'};
+		}
+		$auth_ldap_attr       = $self->{config}->{'auth_ms_ldap_attr'} if ($self->{config}->{'auth_ms_ldap_attr'});
+		$auth_ldap_debug      = NMISNG::Util::getbool($self->{config}->{'auth_ms_ldap_debug'})  if ($self->{config}->{'auth_ms_ldap_debug'});
+		$auth_ldap_group      = $self->{config}->{'auth_ms_ldap_group'} if ($self->{config}->{'auth_ms_ldap_group'});
+		$auth_ldap_privs      = $self->{config}->{'auth_ms_ldap_privs'} if ($self->{config}->{'auth_ms_ldap_privs'});
+		$auth_ldap_privs_file = $self->{config}->{'auth_ms_ldap_privs_file'} if ($self->{config}->{'auth_ms_ldap_privs_file'});
+		$auth_ldap_server     = $self->{config}->{'auth_ms_ldap_server'} if ($self->{config}->{'auth_ms_ldap_server'});
+		$auth_ldaps_server    = $self->{config}->{'auth_ms_ldaps_server'} if ($self->{config}->{'auth_ms_ldaps_server'});
+	}
+
+    if ((!$auth_ldaps_server or $auth_ldaps_server eq "") and (!$auth_ldap_server or $auth_ldap_server eq "")) {
+        NMISNG::Util::logAuth("ERROR Auth::_get_ldap_privs, called but not configured");
+		return 0;
+    }
+
+    if ($auth_ldaps_server and $auth_ldaps_server ne "") {
```

Comment:
this seems to be checking the same thing?

Reply:
Resolved